modules/nw-networkpolicy-about.adoc

```diff
@@ -0,0 +1,100 @@
+// Module included in the following assemblies:
+//
+// networking/configuring-networkpolicy.adoc
+
+[id="nw-networkpolicy-about-{context}"]
+
+= About network policy
+
+In {product-title} {product-version} `NetworkPolicy` is the default mode for the
+`OpenShiftSDN` plug-in for the Kubernetes Container Network Interface (CNI).
+
+[NOTE]
+====
+The `v1` NetworkPolicy features are available only in {product-title}. This
+means that egress policy types, IPBlock, and combining `podSelector` and
+`namespaceSelector` are not available in {product-title}.
+====
+
+In a cluster network isolation is controlled entirely by
+link:https://github.com/kubernetes/community/blob/master/contributors/design-proposals/network/network-policy.md[`NetworkPolicy`
+objects]. By default, all pods in a project are accessible from other pods and
+network endpoints. To isolate one or more pods in a project, you can create
+`NetworkPolicy` objects in that project to indicate the allowed incoming
+connections. Project administrators can create and delete `NetworkPolicy`
+objects within their own project.
+
+Pods that do not have `NetworkPolicy` objects pointing to them are fully
+accessible, whereas, pods that have one or more `NetworkPolicy` objects pointing
+to them are isolated. These isolated pods only accept connections that are
+accepted by at least one of their `NetworkPolicy` objects.
+
+Following are a few sample `NetworkPolicy` object definitions supporting
+different scenarios:
+
+* *Deny All Traffic*
++
+To make a project "deny by default" add a `NetworkPolicy` object that
+matches all pods but accepts no traffic.
++
+[source,yaml]
+----
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: deny-by-default
+spec:
+  podSelector:
+  ingress: []
+----
+
+* *Only Accept connections from pods within project*
++
+To make pods accept connections from other pods in the same project,
+but reject all other connections from pods in other projects:
++
+[source,yaml]
+----
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: allow-same-namespace
+spec:
+  podSelector:
+  ingress:
+  - from:
+    - podSelector: {}
+----
+
+* *Only allow HTTP and HTTPS traffic based on pod labels*
++
+To enable only HTTP and HTTPS access to the pods with a specific label
+(`role=frontend` in following example), add a `NetworkPolicy` object similar to:
++
+[source,yaml]
+----
+kind: NetworkPolicy
+apiVersion: networking.k8s.io/v1
+metadata:
+  name: allow-http-and-https
+spec:
+  podSelector:
+    matchLabels:
+      role: frontend
+  ingress:
+  - ports:
+    - protocol: TCP
+      port: 80
+    - protocol: TCP
+      port: 443
+----
```

Comment:
Definitely add a network policy that shows how to grant access to the OpenShift router.

Reply:
@squeed, is that this or a different policy object?

```yaml
- apiVersion: networking.k8s.io/v1
  kind: NetworkPolicy
  metadata:
    name: allow-from-openshift-ingress-namespace
  spec:
    ingress:
    - from:
      - namespaceSelector:
          matchLabels:
            name: openshift-ingress
    podSelector: {}
    policyTypes:
    - Ingress
```

Reply:
That is right... but I have to get a PR in to the ingress operator to make it actually work! Good think we thought about this.

Reply:
@squeed, I'll comment out the part that references this until it's merged. Thanks!

Reply:
Waiting for https://github.com/openshift/cluster-ingress-operator/pull/218 to merge.